tests/unit/admin/views/test_checks.py

```diff
@@ -177,13 +184,23 @@ def test_sucess(self, db_request):
 
         db_request.task = pretend.call_recorder(lambda *a, **kw: backfill_recorder)
 
-        views.run_backfill(db_request)
-
-        assert db_request.session.flash.calls == [
-            pretend.call(
-                "Running %s on 10000 %ss!" % (check.name, check.hooked_object.value),
-                queue="success",
-            )
-        ]
-
-        assert backfill_recorder.delay.calls == [pretend.call(check.name, 10000)]
+        views.run_evaluation(db_request)
+
+        if check_type == MalwareCheckType.event_hook:
+            assert db_request.session.flash.calls == [
+                pretend.call(
+                    "Running %s on 10000 %ss!"
+                    % (check.name, check.hooked_object.value),
+                    queue="success",
+                )
+            ]
+            assert db_request.task.calls == [pretend.call(backfill)]
+            assert backfill_recorder.delay.calls == [pretend.call(check.name, 10000)]
+        else:
```

Comment:
I'm not sure how I feel about tests for different behavior all in one, since we'll end up with a cascading `elif` here. But for now might `elif check_type == MalwareCheckType.scheduled` and then `else:` with an invalid assert. Would make it easier to tell where the asserts for a given type are and explicitly fail in the implicit case.